x-pack/solutions/security/plugins/security_solution/public/common/components/visualization_actions/use_lens_attributes.tsx

```diff
@@ -34,7 +37,25 @@ export const useLensAttributes = ({
   title,
 }: UseLensAttributesProps): LensAttributes | null => {
   const { euiTheme } = useEuiTheme();
-  const { selectedPatterns, dataViewId, indicesExist } = useSourcererDataView(scopeId);
+  const {
+    selectedPatterns: oldSelectedPatterns,
+    dataViewId: oldDataViewId,
+    indicesExist: oldIndicesExist,
+  } = useSourcererDataView(scopeId);
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+
+  const { dataView } = useDataView(scopeId);
+  const experimentalSelectedPatterns = useSelectedPatterns(scopeId);
+
+  const dataViewId = newDataViewPickerEnabled ? dataView?.id ?? '' : oldDataViewId;
+  const indicesExist = newDataViewPickerEnabled
+    ? !!dataView?.matchedIndices?.length
```

Comment:
Eventually, do we want to wrap this in some utility so like `useHasData(scopeId)`?

Reply:
yes, something like that. maybe we will even skip page loading entirely if the indices are not there?